src/include/sof/audio/audio_stream.h

```diff
@@ -1040,6 +1040,14 @@ static inline void audio_stream_fmt_conversion(enum ipc4_bit_depth depth,
 	*frame_fmt = (depth >> 3) - 2;
 	*valid_fmt = (valid >> 3) - 2;
 
+	if (depth == 8) {
+		*frame_fmt = SOF_IPC_FRAME_U8;
+	}
+
+	if (valid == 8) {
```

Comment:
do you need to check type == IPC4_TYPE_UNSIGNED_INTEGER ?

Reply:
Unfortunately, in my case, the type variable is 0. This enum seems invalid because two values at the same time can be used to describe a type, eg. IPC4_TYPE_MSB_INTEGER and IPC4_TYPE_UNSIGNED_INTEGER.